### ISSUE TITLE
Use docker 1.6 labels for better container ownership

### ISSUE THREAD

**Issue by geophree** [tes/bosco#78]
New in docker 1.6, we can label images and containers

I propose we use this so a bosco knows if it's the one that started a container, so it won't try to stop containers started in other ways.

Label a container with the team or path to team (or repo):
com.tes.bosco=/path/to/something ?
or
com.tes.bosco=team-name/repo-name ?

http://rancher.com/docker-labels/

**Comment by cliftonc** [comment]
It only stops them now if they match the name of a repo doesn't it?

https://github.com/tes/bosco/blob/master/commands/stop.js#L51

But happy to add additional tags if it makes things clearer in Docker land.

**Comment by codeaholics** [comment]
Correct. When Bosco starts a service it will name the container after the service. And it will use the same name to stop the service. 

In fact, I think we currently rely on the behaviour so that Bosco and the makefiles play nicely together and try not to trample on each other.

**Comment by codeaholics** [comment]
https://github.com/tes/module-tsl-resources-build/commit/2680f495eea8814e0acc97fbd1437ca8f10380de

**Comment by codeaholics** [comment]
So if we change this, we have to change the make-world too (perhaps this is the time to have the makefile delegate to Bosco?), but, more importantly, we have to ensure that the first subsequent release of each service is carefully managed to ensure it correctly shuts down the old version it's replacing.

**Comment by geophree** [comment]
I'm mostly thinking we don't want one bosco messing with containers that another bosco (associated with another team) started.  Right now it's only based on container name, I think whichever bosco started a container should own it, and other boscos should just warn about trying to start it when it's already running, or whatnot.

**Comment by cliftonc** [comment]
But teams share containers - most people only want one mongodb or redis is running, and few people other than me work across multiple teams at once.

**Comment by codeaholics** [comment]
Makes sense. Let me explain why I made the change I linked in the makefiles, and then you'll have all the information to make a decision. :-)

If you ran the system locally, Bosco would start all the services including the local infrastructure pieces you need that only run in local (e.g. Redis, Nginx, etc.). If you then ran tests for a service you were working on (e.g. `make good`) that was OK because the makefiles determine which local infra services are already running by trying to connect to them.

However, if you ran tests without having started the system, the makefile would start the local infra services for you (which is A Good Thing™). But if you then tried to start the system, things go weird because Bosco would try and start duplicate copies of those services, etc.

So my change simply makes the containers started by the makefile use the same naming convention as Bosco so Bosco detects them and doesn't try and restart them.

I'm guessing the ideal approach would be to ditch the makefiles totally or at the very least have them use Bosco to start these local development services.

**Comment by cliftonc** [comment]
I'd love to kill the make files and replace then with gulp, this feels like deja vu.  Maybe I'll flesh out my tes gulp library while on the plane next week.
